yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala

```diff
@@ -61,6 +60,12 @@ private[spark] class YarnClientSchedulerBackend(
     // to the executors
     super.start()
 
+    // SPARK-8851: In yarn-client mode, the AM still does the credentials refresh. The driver
+    // reads the credentials from HDFS, just like the executors and updates its own credentials
+    // cache.
+    if (conf.contains("spark.yarn.credentials.file")) {
+      YarnSparkHadoopUtil.get.startExecutorDelegationTokenRenewer(conf)
+    }
     waitForApplication()
```

Comment:
it might make sense to move the startExecutorDelegationTokenRenewer til after waitforApplication since it won't update the credentials until application master is running anyway